clang/tools/CMakeLists.txt

```
@@ -56,3 +56,11 @@ add_clang_subdirectory(libclang)
 
 add_clang_subdirectory(amdgpu-arch)
 add_clang_subdirectory(nvptx-arch)
+
+if ("mlir" IN_LIST LLVM_ENABLE_PROJECTS)
+  if (CLANG_BUILT_STANDALONE)
+    message(WARNING "archgen-mlir doesnt support being build as standalone so it was not enabled")
```

```suggestion
    message(WARNING "archgen-mlir does not support being built as standalone so it was not enabled")
```
But anyway, I have the feeling the logic here should be a hard error and guarded by an option anyway to enable explicitly `archgen-mlir`. For example if someone use this only for AIE or FPGA without ArchGen.

---

It is convenient for me that ArchGenMLIR doesn't have a hard flag. because it allows me not to change my build scripts everywhere.